### ISSUE TITLE
Build failure when building for Renesas GR-PEACH or GR-LYCHEE target

### ISSUE THREAD

**Issue by toyowata** [ARMmbed/mbed-os-example-blinky-baremetal#20]
## Step to reproduce

```shell
$ mbed import mbed-os-example-blinky-baremetal
$ cd mbed-os-example-blinky-baremetal
$ mbed compile -m gr_lychee -t gcc_arm
```

I got link error below (missing `mbed_get_a9_tick_irqn` function for non RTOS build) :

```
Link: mbed-os-example-blinky-baremetal
BUILD/GR_LYCHEE/GCC_ARM/mbed-os/platform/source/SysTimer.o: In function `mbed::internal::SysTimer<1000ul, true>::_set_irq_pending()':
/Users/toywat01/test/renesas/mbed-os-example-blinky-baremetal/./mbed-os/platform/source/SysTimer.cpp:282: undefined reference to `mbed_get_a9_tick_irqn'
BUILD/GR_LYCHEE/GCC_ARM/mbed-os/platform/source/SysTimer.o: In function `mbed::internal::SysTimer<1000ul, true>::_clear_irq_pending()':
/Users/toywat01/test/renesas/mbed-os-example-blinky-baremetal/./mbed-os/platform/source/SysTimer.cpp:298: undefined reference to `mbed_get_a9_tick_irqn'
collect2: error: ld returned 1 exit status
[ERROR] BUILD/GR_LYCHEE/GCC_ARM/mbed-os/platform/source/SysTimer.o: In function `mbed::internal::SysTimer<1000ul, true>::_set_irq_pending()':
/Users/toywat01/test/renesas/mbed-os-example-blinky-baremetal/./mbed-os/platform/source/SysTimer.cpp:282: undefined reference to `mbed_get_a9_tick_irqn'
BUILD/GR_LYCHEE/GCC_ARM/mbed-os/platform/source/SysTimer.o: In function `mbed::internal::SysTimer<1000ul, true>::_clear_irq_pending()':
/Users/toywat01/test/renesas/mbed-os-example-blinky-baremetal/./mbed-os/platform/source/SysTimer.cpp:298: undefined reference to `mbed_get_a9_tick_irqn'
collect2: error: ld returned 1 exit status

[mbed] ERROR: "/Users/toywat01/mbed-venv/bin/python" returned error.
       Code: 1
       Path: "/Users/toywat01/test/renesas/mbed-os-example-blinky-baremetal"
       Command: "/Users/toywat01/mbed-venv/bin/python -u /Users/toywat01/test/renesas/mbed-os-example-blinky-baremetal/mbed-os/tools/make.py -t gcc_arm -m gr_lychee --source . --build ./BUILD/GR_LYCHEE/GCC_ARM"
       Tip: You could retry the last command with "-v" flag for verbose output
---
```

## Mbed OS version

```
mbed-os-example-blinky-baremetal (#e196549b6ddc)
`- mbed-os (#b6e5a0a8afa3, tags: mbed-os-5.14.0, mbed-os-5.14.0-rc3)
```

@ARMmbed/team-renesas

**Comment by ciarmcom** [comment]
*Internal Jira reference: https://jira.arm.com/browse/IOTCORE-1442*

**Comment by toyowata** [comment]
The issue is fixed in the Mbed OS 5.14.2 release.
